### PR TITLE
Tweak the rules for endPrank

### PR DESCRIPTION
**Pull request by anvacaru**
Fixes: #1509 

 - tweak the `#endPrank` rules to avoid overlapping.
 - added a new test for internal prank calls.

There are two differences between the way we inject the pranked address vs the way foundry does it ( https://github.com/foundry-rs/foundry/blob/master/evm/src/executor/inspector/cheatcodes/mod.rs#L380-L395)
     1. We only set the tx.origin at the targeted depth - foundry sets it whenever the `call depth  >= targeted depth`
     2. When setting the msg.caller, we do not check that the current caller is the same as the one who invoked the `startPrank` cheat code.

### CODE REVIEW

File: include/kframework/foundry.md [hunk 1]
```diff
@@ -1145,20 +1147,19 @@ If the production is matched when no prank is active, it will be ignored.
...
+          <active> true </active>
+          ...
+        </prank>
+
+    rule <k> #endPrank => . ... </k>
+        <prank>
+          <active> false </active>
           ...
         </prank>
 ```
```
Comment (ehildenb) [inline include/kframework/foundry.md:1165]: Why would we be ending the prank for `<active> false </active>`? Is this actually a case we expect to be exercised? Do our tests exercise this case?

Comment (ehildenb) [reply include/kframework/foundry.md:1165]: If not, then maybe the first rule is enough on its own?

Comment (anvacaru) [reply include/kframework/foundry.md:1165]: That rule is for the case in which `vm.endPrank()` is invoked when there is no active prank.

Comment (ehildenb) [reply include/kframework/foundry.md:1165]: Is this a case that Foundry allows?

**Review by ehildenb — APPROVED** [review]

File: include/kframework/foundry.md (hunk 1 above)
Comment (anvacaru) [reply include/kframework/foundry.md:1165]: Yes, we have tests for this scenario.

**Pull request merged**